newparp/static/js/newparp.js

```diff
@@ -610,27 +610,31 @@ var msparp = (function() {
 			var latest_date = user.meta.show_timestamps ? new Date(latest_time) : null;
 			var new_messages = [];
 
-			// Websockets
-			var messages_method = typeof(WebSocket) != "undefined" ? "websocket" : "long_poll";
-			var ws_protocol = (location.protocol=="https:") ? "wss://" : "ws://";
+			// Connecting and disconnecting
+
 			var ws;
 			var ws_works = false;
 			var ws_connected_time = 0;
-			function launch_websocket() {
+
+			function connect() {
+				if (typeof(WebSocket) == "undefined") {
+					status_bar.css("color", "#f00").html("Sorry, your browser doesn't appear to support websockets. Please use the latest version of <a href=\"https://www.firefox.com/\">Firefox</a> or <a href=\"https://www.google.com/chrome\">Chrome</a> to participate in this chat.");
+					return;
+				}
+
 				// Don't create a new websocket unless the previous one is closed.
 				// This prevents problems with eg. double clicking the join button.
 				if (ws && ws.readyState != 3) { return; }
 				status = "connecting";
 
-				ws = new WebSocket(ws_protocol + "live." + location.host + "/" + chat.id + "?after=" + latest_message_id);
+				ws = new WebSocket("wss://live." + location.host + "/" + chat.id + "?after=" + latest_message_id);
```

Comment:
Shrug, there's always Let's Encrypt and/or getting your browser to accept a self-signed one.